server/src/test/java/org/elasticsearch/action/update/UpdateRequestTests.java

```diff
@@ -495,6 +496,27 @@ public void testToValidateUpsertRequestWithVersion() {
         assertThat(updateRequest.validate().validationErrors(), contains("can't provide version in upsert request"));
     }
 
+    public void testUpdatingRejectsLongIds() {
+        String id = randomAlphaOfLength(511);
```

Comment:
@wdongyu this can be looped. code isnt compact here and DRY/iteration can be followed.